src/HanziWriter.js

```diff
@@ -178,6 +179,10 @@ HanziWriter.prototype.setCharacter = function(char) {
 
     if (this._options.showCharacter) this._characterRenderer.showImmediate();
     if (this._options.showOutline) this._outlineRenderer.showImmediate();
+  }, () => {
```

Comment:
Could you use the `(...args) => {` style here? babel should essentially turn it into the same result in the end, and it's a bit more readable IMO since it makes it clear this function is expecting arguments.

Reply:
This logic seems like it might make more sense in the error handler you added in `_loadCharacterData()` below. that way all the logic for loading character data and handling errors happens within `_loadCharacterData()`.

Reply:
no prob, I'm a newbie concerning babel and pretty much every thing serious in javascript (I mean outside standard javascript in browsers from a few years ago). Very happy to learn.

I'm happy also to move the logic here to ``_loadCharacterData``, but we have to prevent the then in this first function (``setCharacter()``) to set of : I can't see what I'm supposed to return in the rejected callback at the end of ``_loadCharacterData()`` that won't trigger ``.then()`` from ``setCharacter`` function or complain because of a missing ``reject`` callback in ``setCharacter()`` function.  With this perspective it seems quite natural to me to use the original logic I used as it simply works and I can't see a direct solution with your proposal. See my problem ? I might have missed a simple solution.

Reply:
And as a last connected thought : I saw that many function of HanziWriter are not returning promises where they easily could and this would provide many easy and natural ways to have these callbacks. For instance, why not return the promise created at the end of ``setCharacter()`` ? in this light, it seems very natural to pass along the resolved and the rejected state so as to leave the caller of ``setCharacter()`` to decide if he wants to hook a callback on both failure or success of the resolution of this promise. What do you think of that ?

Reply:
I think it's OK to still trigger then `then()` in `setCharacter`, but maybe we can just have an if-statement at the top to check if there's any data passed in and just exit the function if not? Otherwise we could add a `.catch()` at the end, like you said. It just seems strange that errors in setting up the character would be considered char data loading errors the way it's written currently. I think it's OK to return a promise from `setCharacter()` too if that makes it easier

Reply:
I returned a promise form ``setCharacter()``. Added the condition at the beginning of the ``.then()`` to ignore empty results. And thus use this condition and I am returning null in the ``_loadCharacterData`` rejection callback.